src/assets/scripts/client/ui/SettingsController.js

```diff
@@ -177,20 +177,65 @@ export default class SettingsController {
      * Build the html for a select option.
      *
      * @for SettingsController
-     * @method _buildOptionTemplate
+     * @method _buildSelectOptionTemplate
      * @param optionData {array<string>}
      * @param selectedOption {string}
      * @return optionSelectTempalate {HTML Element}
      * @private
      */
-    _buildOptionSelectTemplate(optionData, selectedOption) {
+    _buildSelectOptionTemplate(optionData, selectedOption) {
         if (optionData.value === selectedOption) {
             return `<option value="${optionData.value}" selected>${optionData.displayLabel}</option>`;
         }
 
         return `<option value="${optionData.value}">${optionData.displayLabel}</option>`;
     }
 
+    /**
+     * Build text input form element
+     *
+     * @for SettingsController
+     * @method _buildTextInputTemplate
+     * @param option {object}
+     * @return $container {jquery Element}
+     */
+    _buildTextInputTemplate(option) {
+        const $container = $(UI_OPTION_CONTAINER_TEMPLATE);
+        const $label = $(UI_OPTION_LABEL_TEMPLATE);
+        const currentValue = GameController.game.option.getOptionByName(option.name);
+        const $selector = $(`<input class="form-input" step="any"
+                                    type="${option.type}"
+                                    name="${option.name}"
+                                    value="${currentValue}">`);
+        const $unit = $(`<span class="form-value">${option.unit}</span>`);
+
+        $container.append($label);
+        $label.text(option.description);
+
+        // TODO: this should be moved to proper event handler method and only assigned here.
+        $selector.change((event) => {
+            const $currentTarget = $(event.currentTarget);
+
+            if (!option.validationHandler($currentTarget.val())) {
```

Comment:
with the `currentValue` variable, we can replace these `$currentTarget.val()` calls